### PR TITLE
Use in-memory pilot in cf/copilot_test

### PR DESCRIPTION
**Pull request by Nino-K**
- This uses the existing pattern of in memory pilot instead of
building and running the binary

### CODE REVIEW

File: tests/e2e/tests/pilot/cloudfoundry/copilot_test.go [hunk 1]
```diff
@@ -21,32 +21,30 @@ import (
...
 	mockmcp "istio.io/istio/tests/e2e/tests/pilot/mock/mcp"
+	"istio.io/istio/tests/util"
 )
 
 const (
 	pilotDebugPort     = 5555
 	pilotGrpcPort      = 15010
-	copilotMCPPort     = 5557
 	sidecarServicePort = 15022
+	mcpServerAddr      = "127.0.0.1:5557"
```
[the remark this answers is not in the document]
Comment (Nino-K) [reply tests/e2e/tests/pilot/cloudfoundry/copilot_test.go]: Thanks for pointing that out, yeah that would be a cleaner approach as the mcp server already returns the port.

**Comment by Nino-K** [comment]
@ayj @rshriram @ZackButcher  can we merge in this PR?

**Comment by ayj** [comment]
Merging into master has been frozen until 1.1 is released. Can you rebase onto 1.1 branch?

**Comment by Nino-K** [comment]
@ayj did you also want me to create the PR against the 1.1 branch?

**Comment by ayj** [comment]
Yes, please :) You should be able to switch the branch for the existing PR by selecting the "edit" button at the top next to the PR title.

**Comment by Nino-K** [comment]
Not too sure why CLA bot is confused, the original commits have not changed and there is also no co-author :thinking:

**Comment by ayj** [comment]
Top-level OWNERs is required due to Gopkg.lock change. @rshriram will need to /lgtm

**Pull request merged**